src/qmeta/QMeta.h

```diff
@@ -58,6 +58,45 @@ class QMeta {
      */
     typedef std::vector<std::pair<std::string, std::string> > TableNames;
 
+    /**
+     * The structure ChunkMap encapsulates a disposition of chunks at Qserv workers
+     * along with a time when the map was updated.
+     *
+     * Here is an example on how to using the map for getting info on all chunks in
+     * the given context:
+     * @code
+     *   std::string const worker   = "worker-001";
+     *   std::string const database = "LSST-DR01";
+     *   std::string const table    = "Object";
+     *
+     *   ChunkMap const& chunkMap = ...;
+     *   for (auto const& [chunk, size] : chunkMap[worker][database][table]) {
+     *       ...
+     *   }
+     * @endcode
+     */
+    struct ChunkMap {
+        /// @return 'true' if the map is empty (or constructed using the default constructor)
+        bool empty() const {
+            return workers.empty() || (std::chrono::time_point<std::chrono::system_clock>() == updateTime);
+        }
+
+        // NOTE: Separate types were added here for the sake of clarity to avoid
+        // a definition of the unreadable nested map.
+
+        typedef std::pair<unsigned int, size_t> Chunk;     ///< Chunk number and its size (bytes)
+        typedef std::vector<Chunk> Chunks;                 ///< Collection of chunks
```

Comment:
To clarify, Chunk<unsigned int = chunk id number, size_t = the size of the database.table in that chunk>. Correct?

Reply:
Yes, that's correct. And since you asked this question (which means the interface is not self-explanatory) I will replace `std::pair` with the following struct:
```
struct ChunkInfo {
    unsigned int chunk = 0;
    size_t size = 0;
};
```